Sources/Core/Types/BuiltinType.swift

```diff
@@ -7,11 +7,20 @@ public enum BuiltinType: TypeProtocol {
   /// and does not specify signedness.
   case i(Int)
 
+  /// A built-in 16-bit floating-point type (specifically, "binary16" in IEEE 754).
+  case half
+
+  /// A built-in 32-bit floating-point type (specifically, "binary32" in IEEE 754).
+  case float
+
   /// A built-in 64-bit floating-point type (specifically, "binary64" in IEEE 754).
-  case f64
+  case double
+
+  /// A built-in 128-bit floating-point type (specifically, "binary128" in IEEE 754).
+  case fp128
 
-  /// A built-in raw pointer pointer.
-  case pointer
+  /// A built-in opaque pointer.
+  case ptr
 
   /// The type of the built-in module.
   case module
```

Comment:
```suggestion
  /// The underlying type of all modules.
  case module
```
?
It would surprise me a bit if there is a special type just for the builtin module but not all others.

Reply:
There's `ModuleType` for all other modules, but it requires a `ModuleDecl`, which doesn't exist for the built-in module specifically.